### ISSUE TITLE
Deployments and Retrieves are not working

### ISSUE THREAD

**Issue by rrosset91** [forcedotcom/cli#2632]
> **Note** 
> Before you submit your issue, make sure that:
> - You're using the latest version of Salesforce CLI.
> - You've searched both open and closed issues for related posts.
> - You've used the `doctor` command to diagnose common issues. 
> - You understand that GitHub Issues don't adhere to any agreement or SLA.
>   - If you require immediate assistance, use official channels such as Salesforce Customer Support.

### Summary
Both from sfdx cli or cli integration extension from VSCODE, the DEPLOY and RETRIEVE actions are resulting in a long time running operation with no error displayed, it just keeps loading, without further action.
Its important to mention that apparently this is happening on all sandboxes of my organization and also using other sources such as Workbench. This morning I tried to do a retrieve from workbench (Since vscode was with this issue) and it took 20 minutes loading and then I just gave up.

### Steps To Reproduce
>On VSCODE:
Right click on any metadata, go to Deploy Source to Org (Or Retrieve Source from Org)
>On WORKBENCH:
Select any package.xml that contains metadata info, and try to deploy it or retrieve it

### Expected result
Source deployed (Or retrieved) successfully

### Actual result
Loading for long time without further actions

### System Information
Windows

```json
PASTE_VERSION_OUTPUT_HERE
```
### Additional information
![image](https://github.com/forcedotcom/cli/assets/29459090/5ca371d8-7c7a-4ef0-8b31-0f1f32266939)

**Comment by rrosset91** [comment]
{
  "architecture": "win32-x64",
  "cliVersion": "@salesforce/cli/2.22.7",
  "nodeVersion": "node-v20.10.0",
  "osVersion": "Windows_NT 10.0.22621",
  "rootPath": "C:\\Users\\RogerRosset\\AppData\\Local\\sf\\client\\2.22.7-453215f",
  "shell": "cmd.exe",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 3.0.3 (core)",
    "@oclif/plugin-commands 3.0.7 (core)",
    "@oclif/plugin-help 6.0.8 (core)",
    "@oclif/plugin-not-found 3.0.5 (core)",
    "@oclif/plugin-plugins 4.1.10 (core)",
    "@oclif/plugin-search 1.0.8 (core)",
    "@oclif/plugin-update 4.1.5 (core)",
    "@oclif/plugin-version 2.0.9 (core)",
    "@oclif/plugin-warn-if-update-available 3.0.6 (core)",
    "@oclif/plugin-which 3.0.12 (core)",
    "@salesforce/cli 2.22.7 (core)",
    "apex 3.0.9 (core)",
    "auth 3.0.12 (core)",
    "data 3.0.6 (core)",
    "deploy-retrieve 2.2.6 (core)",
    "info 3.0.8 (core)",
    "limits 3.0.6 (core)",
    "marketplace 1.0.11 (core)",
    "org 3.1.2 (core)",
    "packaging 1.27.10 (core)",
    "schema 3.0.8 (core)",
    "settings 2.0.9 (core)",
    "sobject 1.0.6 (core)",
    "source 3.0.3 (core)",
    "telemetry 3.1.6 (core)",
    "templates 56.0.4 (core)",
    "trust 3.2.0 (core)",
    "user 3.1.1 (core)"
  ]
}

**Comment by 1ktribble** [comment]
This is also occurring for our team and we all have MacOS devices. This command uses the old `sfdx force:source:deploy` command instead of the new `sf project deploy start` command. When I tried using `sf project deploy start` instead, I received a different error about `Unexpected child metadata ... found for parent type [CustomObject]` for custom metadata records.

Our timeout is set to 10 minutes, and at times, deploying a single class surpasses that timeout limit. Prior to v59.0 of the SOAP API, deployments were near-instantaneous.

```
{
  "architecture": "darwin-arm64",
  "cliVersion": "@salesforce/cli/2.21.8",
  "nodeVersion": "node-v18.16.0",
  "osVersion": "Darwin 22.6.0",
  "rootPath": "/usr/local/lib/node_modules/@salesforce/cli",
  "shell": "zsh",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 3.0.2 (core)",
    "@oclif/plugin-commands 3.0.7 (core)",
    "@oclif/plugin-help 6.0.7 (core)",
    "@oclif/plugin-not-found 3.0.4 (core)",
    "@oclif/plugin-plugins 4.1.9 (core)",
    "@oclif/plugin-search 1.0.8 (core)",
    "@oclif/plugin-update 4.1.4 (core)",
    "@oclif/plugin-version 2.0.8 (core)",
    "@oclif/plugin-warn-if-update-available 3.0.5 (core)",
    "@oclif/plugin-which 3.0.11 (core)",
    "@salesforce/cli 2.21.8 (core)",
    "apex 3.0.8 (core)",
    "auth 3.0.10 (core)",
    "data 3.0.5 (core)",
    "deploy-retrieve 2.2.3 (core)",
    "info 3.0.7 (core)",
    "limits 3.0.5 (core)",
    "marketplace 1.0.10 (core)",
    "org 3.1.1 (core)",
    "packaging 1.26.4 (user)",
    "schema 3.0.8 (core)",
    "settings 2.0.9 (core)",
    "sobject 1.0.5 (core)",
    "source 3.0.3 (core)",
    "telemetry 3.1.4 (core)",
    "templates 56.0.4 (core)",
    "trust 3.1.2 (core)",
    "user 3.1.0 (core)",
    "sfdx-git-delta 5.31.1 (user)",
    "sfpowerkit 6.1.0 (user)",
    "texei-sfdx-plugin 2.1.3 (user)"
  ]
}
```

**Comment by cristiand391** [comment]
I'm going to close this as a dup of https://github.com/forcedotcom/cli/issues/2631
